File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedintegration/MainActivity.java
package com.firewok.example.java_integration.feedintegration;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.firewok.example.java_integration.feedintegration.activity.ActivityIntegrationActivity;
import com.firewok.example.java_integration.feedintegration.addremove.AddRemoveIntegrationActivity;
import com.firewok.example.java_integration.databinding.ActivityMainBinding;
import com.firewok.example.java_integration.feedintegration.fragment.FragmentIntegrationActivity;
import com.firewok.example.java_integration.feedintegration.recyclerview.RecyclerViewIntegrationActivity;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.activityFeed.setOnClickListener(v -> {
            Intent intent = ActivityIntegrationActivity.getIntent(this);
            startActivity(intent);
        });

        binding.fragmentFeed.setOnClickListener(v -> {
            Intent intent = FragmentIntegrationActivity.getIntent(this);
            startActivity(intent);
        });

        binding.recyclerViewFeed.setOnClickListener(v -> {
            Intent intent = RecyclerViewIntegrationActivity.getIntent(this);
            startActivity(intent);
        });

        binding.addRemoveFeed.setOnClickListener(v -> {
            Intent intent = AddRemoveIntegrationActivity.getIntent(this);
            startActivity(intent);
        });
    }
}

File: feed_integration/app/src/main/java/tv/fw/feedintegration/MainActivity.kt
package tv.fw.feedintegration

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import tv.fw.feedintegration.activity.ActivityIntegrationActivity
import tv.fw.feedintegration.addremove.AddRemoveIntegrationActivity
import tv.fw.feedintegration.databinding.ActivityMainBinding
import tv.fw.feedintegration.fragment.FragmentIntegrationActivity
import tv.fw.feedintegration.recyclerview.RecyclerViewIntegrationActivity

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)

        binding.activityFeed.setOnClickListener {
            startActivity(ActivityIntegrationActivity.intent(this))
        }
        binding.fragmentFeed.setOnClickListener {
            startActivity(FragmentIntegrationActivity.intent(this))
        }
        binding.recyclerViewFeed.setOnClickListener {
            startActivity(RecyclerViewIntegrationActivity.intent(this))
        }
        binding.addRemoveFeed.setOnClickListener {
            startActivity(AddRemoveIntegrationActivity.intent(this))
        }
    }
}
